classes/fields/boolean.php

```diff
@@ -174,14 +174,19 @@ public function data( $name, $value = null, $options = null, $pod = null, $id =
 	 * {@inheritdoc}
 	 */
 	public function validate( $value, $name = null, $options = null, $fields = null, $pod = null, $id = null, $params = null ) {
+		$return = parent::validate( $value, $name, $options, $fields, $pod, $id, $params );
 
 		if ( ! $this->is_required( $options ) ) {
 			// Any value can be parsed to boolean.
-			return true;
+			return $return;
 		}
 
 		$errors = array();
-		$check  = $this->pre_save( $value, $id, $name, $options, $fields, $pod, $params );
+		if ( is_array( $return ) ) {
+			$errors = $return;
```

Comment:
In cases where errors are presented by the parent method, the methods in the field type classes override the whole `$errors` when running their own validation with failure. Is this intentional or should it merge arrays?

Reply:
No that isn't happening. If the parent method results in errors the `$errors` var starts off with those errors. Any extra errors will be appended.